### ISSUE TITLE
Memory leak occurs even after calling dispose

### ISSUE THREAD

**Issue by MinSeungHyun** [ryanheise/just_audio#1355]
**Which API doesn't behave as documented, and how does it misbehave?**
Some instances are still in memory even after calling dispose().

**Minimal reproduction project**
It can be reproduced in all the examples using AudioPlayer.
https://github.com/MinSeungHyun/just_audio_example/blob/ec0be9b7e76f15fb7926fdba07cc861621f663b6/lib/main.dart#L30-L50

**To Reproduce (i.e. user steps, not code)**
1. Play an audio multiple times. (but by creating a new AudioPlayer instance and disposing it each time)
2. You can see `BehaviorSubject`, `_Wrapper`, `DefferStream` and some other instances are still remaining in memory.

**Expected behavior**
All the instances should be freed after calling dispose().

**Screenshots**
The video includes audio.

https://github.com/user-attachments/assets/8423050b-78c7-4940-ae3a-e34784287f74

**Flutter SDK version**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.24.4, on macOS 15.1 24B83 darwin-arm64, locale en-KR)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2024.2.3)
[✓] VS Code (version 1.94.0)
[✓] Connected device (4 available)
[✓] Network resources

• No issues found!
```

**Additional context**
Actually, I found this problem a year ago and fixed it in my repo. I've been using the fixed repo well in my application.

**Comment by ryanheise** [comment]
Thanks for contributing! It is now merged.

**Comment by MinSeungHyun** [comment]
Thanks! Can I know when it will be released?

**Comment by ryanheise** [comment]
I've just done a few merges into `minor` - would you be able to confirm that `minor` works on your end? If so, I'll go ahead and publish a release.

**Comment by MinSeungHyun** [comment]
I've tested on [this commit](https://github.com/MinSeungHyun/just_audio_example/commit/739b5ad7f2bcd9ef1d599a37c11ff1ba145f3770) and my application. It's working well.

**Comment by ryanheise** [comment]
Since it is the `minor` branch that will be released, would you be able to confirm that this specifically is working for you as well?

**Comment by MinSeungHyun** [comment]
Yes, it's the latest minor branch that I tested, mentioned above.
https://github.com/MinSeungHyun/just_audio_example/commit/739b5ad7f2bcd9ef1d599a37c11ff1ba145f3770#diff-8b7e9df87668ffa6a04b32e1769a33434999e54ae081c52e5d943c541d4c0d25L18

**Comment by ryanheise** [comment]
The HEAD of `minor` is 3d3815f4eabc54daf316bca8d020ed201952a42c

But regardless, I have gone ahead and published the new release, so hopefully there won't be any issues.

**Comment by MinSeungHyun** [comment]
The `resolved-ref` here is 3d3815f4eabc54daf316bca8d020ed201952a42c

![Screenshot_20241030_213626_GitHub.jpg](https://github.com/user-attachments/assets/40928c85-b6f1-48b4-a75e-cf96d1730668)
